bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/type/ModuleTypeManager.java

```diff
@@ -0,0 +1,205 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core.type;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Locale;
+
+import org.eclipse.smarthome.automation.handler.provider.ModuleTypeProvider;
+import org.eclipse.smarthome.automation.type.ActionType;
+import org.eclipse.smarthome.automation.type.CompositeActionType;
+import org.eclipse.smarthome.automation.type.CompositeConditionType;
+import org.eclipse.smarthome.automation.type.CompositeTriggerType;
+import org.eclipse.smarthome.automation.type.ConditionType;
+import org.eclipse.smarthome.automation.type.ModuleType;
+import org.eclipse.smarthome.automation.type.TriggerType;
+import org.osgi.framework.BundleContext;
+import org.osgi.util.tracker.ServiceTracker;
+
+/**
+ * @author Yordan Mihaylov - Initial Contribution
+ */
+public class ModuleTypeManager {
```

Comment:
1) Yes it is same as TemplateManager. The Manager classes carries the common implementation of ModuleTypeAdmin and ModuleTypeRegistry.
2) we have already removed setters form ModuleType and the ModuleType becomes immutable object. Method createCopy is removed. We have to do same with Template interface. The change will be available on next delivery.
3) Ok, but Marin has to comment too.
4) OK